src/io/observe.cc

```diff
@@ -316,19 +319,21 @@ void Observe::compute_field_values(Field<spacedim, Value> &field)
 
     OutputDataFieldMap::iterator it=observe_field_values_.find(field.name());
     if (it == observe_field_values_.end()) {
-        observe_field_values_[field.name()] = std::make_shared< OutputData<Value> >(field, points_.size());
+        observe_field_values_[field.name()]
+					= std::make_shared< ElementDataCache<ElemType> >(field.name(), (unsigned int)Value::NRows_,
+			        		(unsigned int)Value::NCols_, points_.size());
         it=observe_field_values_.find(field.name());
     }
-    OutputData<Value> &output_data = dynamic_cast<OutputData<Value> &>(*(it->second));
+    ElementDataCache<ElemType> &output_data = dynamic_cast<ElementDataCache<ElemType> &>(*(it->second));
 
     unsigned int i_data=0;
     for(ObservePoint &o_point : points_) {
         unsigned int ele_index = o_point.element_idx_;
         const Value &obs_value =
-                Value( const_cast<typename Value::return_type &>(
-                        field.value(o_point.global_coords_,
-                                ElementAccessor<spacedim>(this->mesh_, ele_index,false)) ));
-        output_data.store_value(i_data,  obs_value);
+                        Value( const_cast<typename Value::return_type &>(
+                                field.value(o_point.global_coords_,
+                                        ElementAccessor<spacedim>(this->mesh_, ele_index,false)) ));
+        output_data.store_value(i_data,  obs_value.mem_ptr());
```

Comment:
add assert that output_data single element size match size of obs_value, i.e. obs_value.n_rows()*obs_value.n_cols().

Or at least add comment. Simliarly in register field values in OutputTime.